### ISSUE TITLE
No apparent way to browse resolved discussions

### ISSUE THREAD

**Issue by lilyball** [Reviewable/Reviewable#291]
I don't see any way to browse through resolved discussions. If I set my diff bounds appropriately and go through the diff I can see the very tiny checkmark icons and click them to view the associated discussion, but I would have expected there to be some way to view a list of all discussions associated with a review (e.g. how GitHub's "Conversation" tab for a PR shows all code comments).

The context here is I wanted to refer to a discussion I had recently acknowledged but it wasn't obvious how to find it again.

**Comment by pkaminski** [comment]
Yes, this is a variation on #94 and #136, on the general theme of comment visibility control.  I have a design in mind for fixing all of these at once but it's not at the top of the priority list at the moment.  As a somewhat crappy workaround, you can find the comment in the corresponding GitHub message in the PR, and click on the link to see it expanded in Reviewable.

**Comment by kballard** [comment]
Glad to hear you're already thinking about the problem :) And I'm also happy that there is a workaround. I actually haven't looked at the GitHub PR for any PR that I've been using Reviewable for, so I hadn't noticed the list of comment links, so it's good to be aware of that.

**Comment by erikwright** [comment]
I have a review that had two discussions, since resolved. My colleague was incredibly confused because he couldn't see the comments anywhere when he returned to look at the PR. He thought he was looking at a new, very similar PR.

I also looked with him and later at my own desk and couldn't figure it out. After about 15 minutes I have finally found those little checkmarks in the left-hand margin.

I would vote for bumping the priority of this one.

**Comment by porcus** [comment]
I too would like to have a way to browse discussions that have already been marked as resolved.  In some cases, other participants in the review may have had (already resolved) questions/concerns/observations on the review of which I'd like to be aware but that I haven't seen yet.  But more often, what happens is that I marked a discussion as resolved, but then I want to be able to locate it again, so that I can review something said by someone in it.  When there are many files in a PR/review, this becomes particularly arduous.

Ideally, I'd really like for reviewable to have:

1. a discussion view (grouped by file), where discussions (and their current status) can be reviewed.  
2. means to search the content of all discussions, including those which are resolved.

As prominently a role as discussions can play in the review process, it would be nice to see a few future improvements with respect to managing discussions in all states.  In the meantime, I'll try to use the recommended workaround.

**Comment by pkaminski** [comment]
There's also another trick you can use to expand all discussions that I've mentioned in the past but can't find any more, so I'll record it here:

Click on the small dropdown button next to any "Show N older comments" in a discussion header, then click on "Show all in all files".  This will reveal all discussions and force them to expand fully.

But yes, better comment control is still on the todo list, and will probably include a "flattened" discussion list.

**Comment by dkeitel** [comment]
> then click on "Show all in all files"

Would it be possible to expose this function to the custom keybindings? That would be quite useful for us while better comment control is being worked on.

**Comment by pkaminski** [comment]
I added a `showHideAllComments` command for binding.

**Comment by dkeitel** [comment]
Awesome thanks!

**Comment by porcus** [comment]
Much appreciated!

**Comment by dhbradshaw** [comment]
@badz0 : I think we can resolve this by adding one more key binding: a showHideAllDiscussions command.

Here are notes from Piotr:

```
[adding a showHideAllDiscussions] should be pretty straightforward.  You can see how to reveal a group of discussions in src/view/reviews/diff/resolved-marker.vue, and there are actually two methods to reveal top-level and line-level (per file) locators:

review.discussions.top.locators.reveal()
review.files[fileKey].diff.current.locators.reveal()
```

**Comment by pkaminski** [comment]
FWIW, revealing is a one-way operation so `revealAllDiscussions` or `showAllDiscussions` would be a better name for the bindable command.

**Comment by a-poliusov** [comment]
guys, please provide a explanation on how to execute the commands you made @pkaminski

**Comment by pkaminski** [comment]
There are two ways to force all comments to show:
1. You can bind a keyboard shortcut to `showHideAllComments(true)` following the instructions in [the docs](https://docs.reviewable.io/accountsettings.html#custom-key-bindings).  The command has a placeholder in the [sample bindings file](https://reviewable.io/bindings.json) so you'd just need to add the shortcut and install it in your settings.
2. If you don't need to do this regularly, you can instead open up the JS console in the browser and execute `Reviewable.showAllComments()` (and `Reviewable.hideAllComments()` to hide them).
